pkg/kube/controllers/extendedsecret/extendedsecret_reconciler_test.go

```diff
@@ -333,18 +333,24 @@ var _ = Describe("ReconcileExtendedSecret", func() {
 				esv1.LabelKind: esv1.GeneratedSecretKind,
 			}
 
-			client.CreateCalls(func(context context.Context, object runtime.Object, _ ...crc.CreateOption) error {
-				secret := object.(*corev1.Secret)
-				Expect(secret.StringData["password"]).To(Equal(password))
-				Expect(secret.GetName()).To(Equal("generated-secret"))
-				Expect(secret.GetLabels()).To(HaveKeyWithValue(esv1.LabelKind, esv1.GeneratedSecretKind))
+			client.UpdateCalls(func(context context.Context, object runtime.Object, _ ...crc.UpdateOption) error {
+				switch object.(type) {
+				case *esv1.ExtendedSecret:
```

Comment:
Like staticcheck suggest:

pkg/kube/controllers/extendedsecret/extendedsecret_reconciler_test.go:337:12: assigning the result of this type assertion to a variable (switch object := object.(type)) could eliminate the following type assertions:

Reply:
Done.